.github/workflows/validate.yml

```diff
@@ -0,0 +1,74 @@
+name: Validate
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  typecheck:
+    name: ʦ TypeScript (${{ matrix.project }})
+
+    strategy:
+      fail-fast: false
+
+      matrix:
+        project: [blob, fetch, file, form-data, stream]
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: ⬇️ Checkout repo
+        uses: actions/checkout@v3
+
+      - name: ⎔ Setup node
+        uses: actions/setup-node@v3
+        with:
+          cache: yarn
+          node-version: 16
+
+      - name: 📥 Install deps
+        run: yarn --frozen-lockfile
+
+      - name: 🔎 Type check
+        run: yarn typecheck
+        working-directory: packages/${{matrix.project}}
+
+  test:
+    name: 🧪 Test (${{ matrix.project }} / ${{ matrix.os }} / node@${{ matrix.node }})
+
+    strategy:
+      fail-fast: false
+
+      matrix:
+        node-version: [14, 16, 18, 20]
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        project: [blob, fetch, file, form-data, stream]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: ⬇️ Checkout repo
+        uses: actions/checkout@v3
+
+      - name: ⎔ Setup node
+        uses: actions/setup-node@v3
+        with:
+          cache: yarn
+          node-version: ${{ matrix.node-version }}
+
+      - name: 📥 Install deps
+        run: yarn --frozen-lockfile
+
+      - name: 🧪 Test (CJS)
+        run: yarn test:cjs
+        working-directory: packages/${{matrix.project}}
+
+      - name: 🧪 Test (ES)
+        run: yarn test:es
```

Comment:
I don't think `packages/fetch` has a `test:es` or `test:web`?  I wonder if it's better to just run `yarn:test` once in the workflow and let the individual packages compose them together however necessary?

Reply:
`fetch` has a ESM + CJS test

https://github.com/remix-run/web-std-io/blob/5a3703c5c17523b9e8d39ad05eb58c3acc1734c3/.github/workflows/fetch.yml#L78-L79
https://github.com/remix-run/web-std-io/blob/5a3703c5c17523b9e8d39ad05eb58c3acc1734c3/.github/workflows/fetch.yml#L88-L89

I like the different steps though, so maybe we should just make sure that each package has all 3 scripts and just exit early on `test:web` for `fetch`?

Reply:
Or just use `npm run test:web --if-present` 🤷‍♂️

Reply:
> I also think that means we need 65 VM instances/workers per PR which has got to carry some overhead?

Steps are not counted towards checks as you can have multiple steps for each check

Reply:
ah ok, I thought that was a jump from the prior numbers but I see we were already running that many.

I personally still prefer composing `test:*` in `package.json` but I see this is how they were split out before so if we decided to tackle that it could come in a separate PR 👍